packages/wonder-blocks-modal/src/components/flexible-dialog.tsx

```diff
@@ -0,0 +1,205 @@
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+import type {PropsFor, StyleType} from "@khanacademy/wonder-blocks-core";
+
+import {Id} from "@khanacademy/wonder-blocks-core";
+import {breakpoint} from "@khanacademy/wonder-blocks-tokens";
+import {Heading} from "@khanacademy/wonder-blocks-typography";
+import ModalDialog from "./modal-dialog";
+import FlexiblePanel from "./flexible-panel";
+
+/**
+ * The visible title content for the dialog. An ID will be automatically applied
+ * for aria-labelledby on the dialog to the heading content or a div wrapper.
+ */
+type TitleAndAriaLabelledBy = {
+    title: React.ReactNode;
+    "aria-label"?: never;
+    "aria-labelledby"?: string;
+};
+/**
+ * The optional accessible name of the modal.
+ * This is useful when there is no main heading in the dialog. A visible
+ * heading with an id is preferred for automatic matching with aria-labelledby.
+ */
+type AriaLabelOnly = {
+    title?: never;
+    "aria-label": string;
+    "aria-labelledby"?: never;
+};
+/**
+ * The optional ID of the content describing this dialog, if applicable.
+ */
+type AriaLabelledByOnly = {
+    title?: never;
+    "aria-label"?: never;
+    "aria-labelledby": string;
+};
+
+/**
+ * One of the labeling methods is required:
+ */
+type AccessibleLabelProps =
+    | TitleAndAriaLabelledBy
+    | AriaLabelOnly
+    | AriaLabelledByOnly;
+
+type Props = AccessibleLabelProps & {
+    /**
+     * The main heading of the modal, for labeling the dialog.
+     * If string content, an ID will be generated for the heading for aria-labelledby on the dialog.
+     * If a node, an ID will be applied to a DIV wrapping the node.
+     */
+    title?: React.ReactElement | string;
```

Comment:
Does the `title` prop need to be redefined here? Would it override/conflict with how `title` is configured as part of `AccessibleLabelProps`? 

Types tests can help verify this! Related comment: https://github.com/Khan/wonder-blocks/pull/2717/files#r2223077582

Reply:
No, it shouldn't have been redefined... I pushed an update and fiddled with the types a bit!